toolkit/tools/scheduler/schedulerutils/buildworker.go

```diff
@@ -186,6 +189,35 @@ func getBuildDependencies(node *pkggraph.PkgNode, pkgGraph *pkggraph.PkgGraph, g
 	return
 }
 
+func parseCheckSection(logFile string) (checkErr error) {
+	file, logErr := os.Open(logFile)
+	// If we can't open the log file, that's a build error.
+	if logErr != nil {
+		logger.Log.Errorf("Failed to open log file '%s' while checking package test results. Error: %v", logFile, logErr)
+		checkErr = logErr
+		return
+	}
+	defer file.Close()
+	for scanner := bufio.NewScanner(file); scanner.Scan(); {
+		currLine := scanner.Text()
+		// Anything besides 0 is a failed test
+		if strings.Contains(currLine, "CHECK DONE") && !strings.Contains(currLine, "EXIT STATUS 0") {
+			failedLogFile := fmt.Sprintf("%s-FAILED_TEST-%d.log", logFile[:strings.Index(logFile, ".log")], time.Now().UnixMilli())
+			checkErr = os.Rename(logFile, failedLogFile)
+			if checkErr != nil {
+				logger.Log.Errorf("Log file rename failed. Error: %v", checkErr)
+				return
+			}
+			checkErr = fmt.Errorf("Package test failed. Test status line: %s", currLine)
```

Comment:
We try to follow conventions suggested by `go-staticcheck`, which in this case asks to not begin errors with upper case. And to not add a period at the end, but that's not a problem here.
```suggestion
			checkErr = fmt.Errorf("package test failed. Test status line: %s", currLine)
```

If you're using VS Code, then I think [this plug-in](https://marketplace.visualstudio.com/items?itemName=golang.Go) should help with spotting these suggestions.